src/sidebar/services/annotations.ts

```diff
@@ -16,6 +16,21 @@ import type { SidebarStore } from '../store';
 import type { AnnotationActivityService } from './annotation-activity';
 import type { APIService } from './api';
 
+/**
+ * Wrap all occurrences of @mentions in provided text into the corresponding
+ * special tag, as long as they are surrounded by "empty" space (space, tab, new
+ * line, or beginning/end of the whole text).
+ *
+ * For example: `@someuser` with the `hypothes.is` authority would become
+ *  `<a data-hyp-mention data-userid="acct:someuser@hypothes.is">@someuser</a>`
+ */
+function mentionsToTags(text: string, authority: string): string {
+  return text.replace(/(?:^|\s)@(\w+)(?=\s|$)/g, (match, username) => {
+    const userid = `acct:${username}@${authority}`;
+    return ` <a data-hyp-mention data-userid="${userid}">@${username}</a>`;
```

Comment:
The safer way to do this that handles escaping would be to construct a DOM node, set attributes and then deserialize it using `outerHTML`.